packages/dx-react-chart-demos/src/demo-sources/export/bootstrap4/demo.jsx

```diff
@@ -0,0 +1,283 @@
+    .find(node => node.dataset[Symbol('animation')]);
+  if (!style) {
+    style = document.createElement('style');
+    style.type = 'text/css';
+    style.dataset[Symbol('animation')] = true;
+    head.appendChild(style);
+  }
+  const content = style.textContent;
+  if (!content.includes(name)) {
+    style.textContent += `\n@keyframes ${name} ${def}\n`;
+  }
+};
+
+const getLabelAnimationName = () => {
+  const name = 'animation_label_opacity';
+  addKeyframe(name, '{ 0% { opacity: 0; } 99% { opacity: 0; } 100% { opacity: 1; } }');
+  return name;
+};
+
+const Export = () => {
+  const [anchorEl, setAnchorEl] = useState(null);
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+  const handleClick = (event) => {
+    if (!anchorEl) {
+      setAnchorEl(event.currentTarget);
+    } else {
+      handleClose();
+    }
+  };
+
+  const handleExport = option => () => {
+    const filter = node => (node.id !== 'iconButton');
+    const chart = document.body.childNodes[4];
+    const width = chart.offsetWidth;
+    const height = chart.offsetHeight;
+    switch (option) {
+      case 'JPEG':
+        domtoimage.toJpeg(chart, { filter })
+          .then((dataUrl) => {
+            const link = document.createElement('a');
+            link.download = 'chart.jpeg';
+            link.href = dataUrl;
+            link.click();
+          });
+        break;
+      case 'PNG':
+        domtoimage.toPng(chart, { filter })
+          .then((dataUrl) => {
+            const link = document.createElement('a');
+            link.download = 'chart.png';
+            link.href = dataUrl;
+            link.click();
+          });
+        break;
+      case 'PDF':
+        domtoimage.toJpeg(chart, { filter })
+          .then((dataUrl) => {
+            const doc = new JsPDF({
+              orientation: 'landscape',
+              unit: 'px',
+              format: [width, height],
+            });
+            const pdfWidth = doc.internal.pageSize.getWidth();
+            const pdfHeight = doc.internal.pageSize.getHeight();
+            doc.addImage(dataUrl, 'JPEG', 0, 0, pdfWidth, pdfHeight);
+            doc.save('chart');
+          });
+        break;
+      default:
+        domtoimage.toJpeg(chart, { filter })
+          .then((dataUrl) => {
+            let html = '<html><head><title></title></head>';
+            html += '<body style="width: 100%; padding: 0; margin: 0;"';
+            html += ' onload="window.focus(); window.print(); window.close()">';
+            html += `<img src="${dataUrl}" /></body></html>`;
+
+            const printWindow = window.open('', 'print');
+            printWindow.document.open();
+            printWindow.document.write(html);
+            printWindow.document.close();
+          });
+    }
+    handleClose();
+  };
+  return (
+    <Plugin name="Export">
+      <Template name="top">
+        <TemplatePlaceholder />
+        <button
+          type="button"
```

Comment:
Is it required?

Reply:
Yes. There is a warning: missing an explicit type attribute for button eslint(react/button-has-type)